### ISSUE TITLE
sql: truncating pre-2.1 tables cause orphan sequence references

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#47267]
tldr: tables using sequence created in v2.0 or before can cause problem when used in cluster running v2.1+

If a table  uses a sequence (e.g. `DEFAULT nextval('s')` this relationship is recorded in the sequence descriptor, so that DROP SEQUENCE CASCADE also drops the table.

However, when the table changes IDs, such as during TRUNCATE, this relationship must be updated as well. The code to do so was introduced in v2.1, but relies on a descriptor field `UsesSequenceIds` which is only populated for tables created in v2.1 or later.

So the following scenario is problematic:

1. in v2.0 create sequence `s`, then table `t` using `s`
2. upgrade to recent version
3. TRUNCATE the table (this changes the table ID)
4. observe: the sequence is unliked, and DROP SEQUENCE makes the table invalid.

**Comment by arulajmani** [comment]
> If a table uses a sequence (e.g. DEFAULT nextval('s') this relationship is recorded in the sequence descriptor, so that DROP SEQUENCE CASCADE also drops the table.

An aside note on this -- `DROP SEQUENCE CASCADE` shouldn't drop the table, the PG behavior is to simply unlink the sequence dependency from the column and drop the sequence (I find the wording on their docs sort of confusing, but I've tried this out locally). We currently don't support `DROP SEQUENCE CASCADE` (unless the syntax is a no-op) -- so if a sequence is used by any table at all, we simply can't drop it. 

> observe: the sequence is unliked, and DROP SEQUENCE makes the table invalid.
Could I get some clarification on this please? If there is no `UsesSequenceIDs` field, how does the sequence get unlinked? 

I wonder if we can just close this issue now that we have stable Table IDs even during truncates because of #52235.

**Comment by knz** [comment]
yes this issue is moot with the new truncate code
